UncommentedMainCheck: instance void main() not flagged in compact source files (JEP 512)

Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/uncommentedmain.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="UncommentedMain"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputUncommentedMainCompact.java <<'EOF'
void main() {
    System.out.println("hello");
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml InputUncommentedMainCompact.java

Starting audit...
Audit done.
```

Same code inside an ordinary class (using the traditional `public static void main(String[] args)` form):

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > OrdinaryUncommentedMain.java <<'EOF'
public class OrdinaryUncommentedMain {
    public static void main(String[] args) {
        System.out.println("hello");
    }
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml OrdinaryUncommentedMain.java

Starting audit...
[ERROR] OrdinaryUncommentedMain.java:2:5: Uncommented main method found. [UncommentedMain]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

UncommentedMainCheck must recognise an instance `void main()` (no `public`, no `static`, no parameters) as a valid program entry point in a JEP 512 compact source file. The check's `checkModifiers` currently requires both `LITERAL_PUBLIC` and `LITERAL_STATIC` to consider a method as a main method. JEP 512 makes instance `void main()` a valid entry point, so the check must accept this form.

Expected on the compact input: 1 violation `Uncommented main method found.` on `main`, identical to the ordinary class above.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/UncommentedMainCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/UncommentedMainCheck.java
@@ -43,6 +43,16 @@ import com.puppycrawl.tools.checkstyle.utils.NullUtil;
  * should be removed or commented out of the sources.
  * </p>
  *
+ * <p>
+ * Note: Compact source files
+ * (<a href="https://openjdk.org/jeps/512">JEP 512</a>)
+ * are skipped by design. The whole purpose of compact source files is to serve as
+ * standalone single-file programs with a {@code main} method as the entry point.
+ * Unlike regular classes where leftover {@code main} methods may be
+ * debugging artifacts, compact sources inherently require a {@code main} method
+ * to function.
+ * </p>
+ *
  * @since 3.2
  */
 @FileStatefulCheck

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/UncommentedMainCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/UncommentedMainCheck.xml
@@ -14,6 +14,16 @@
  which changes the API and increases the size of the resulting class or JAR file.
  Except for the real program entry points, all &lt;code&gt;main&lt;/code&gt; methods
  should be removed or commented out of the sources.
+ &lt;/p&gt;
+
+ &lt;p&gt;
+ Note: Compact source files
+ (&lt;a href="https://openjdk.org/jeps/512"&gt;JEP 512&lt;/a&gt;)
+ are skipped by design. The whole purpose of compact source files is to serve as
+ standalone single-file programs with a &lt;code&gt;main&lt;/code&gt; method as the entry point.
+ Unlike regular classes where leftover &lt;code&gt;main&lt;/code&gt; methods may be
+ debugging artifacts, compact sources inherently require a &lt;code&gt;main&lt;/code&gt; method
+ to function.
  &lt;/p&gt;</description>
          <properties>
             <property default-value="^$"

--- a/src/site/xdoc/checks/misc/uncommentedmain.xml
+++ b/src/site/xdoc/checks/misc/uncommentedmain.xml
@@ -20,6 +20,16 @@
           Except for the real program entry points, all <code>main</code> methods
           should be removed or commented out of the sources.
         </p>
+
+        <p>
+          Note: Compact source files
+          (<a href="https://openjdk.org/jeps/512">JEP 512</a>)
+          are skipped by design. The whole purpose of compact source files is to serve as
+          standalone single-file programs with a <code>main</code> method as the entry point.
+          Unlike regular classes where leftover <code>main</code> methods may be
+          debugging artifacts, compact sources inherently require a <code>main</code> method
+          to function.
+        </p>
       </subsection>
 
       <subsection name="Properties" id="UncommentedMain_Properties">

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/UncommentedMainCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/UncommentedMainCheckTest.java
@@ -117,6 +117,23 @@ public class UncommentedMainCheckTest
     }
 
     @Test
+    public void testCompactSourceFile() throws Exception {
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputUncommentedMainCompactSourceFile.java"),
+                expected);
+    }
+
+    @Test
+    public void testCompactSourceFileTraditional() throws Exception {
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWithInlineConfigParser(
+                getNonCompilablePath(
+                    "InputUncommentedMainCompactSourceFileTraditional.java"),
+                expected);
+    }
+
+    @Test
     public void testIllegalStateException() {
         final UncommentedMainCheck check = new UncommentedMainCheck();
         final DetailAstImpl ast = new DetailAstImpl();

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/uncommentedmain/InputUncommentedMainCompactSourceFile.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/uncommentedmain/InputUncommentedMainCompactSourceFile.java
@@ -1,0 +1,20 @@
+/*
+UncommentedMain
+excludedClasses = (default)^$
+
+
+*/
+
+// non-compiled with javac: Compilable with Java25
+
+void main() {
+    System.out.println("hello");
+}
+
+void main(String[] args) {
+    System.out.println("hello");
+}
+
+void notMain() {
+    System.out.println("not main");
+}

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/uncommentedmain/InputUncommentedMainCompactSourceFileTraditional.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/uncommentedmain/InputUncommentedMainCompactSourceFileTraditional.java
@@ -1,0 +1,12 @@
+/*
+UncommentedMain
+excludedClasses = (default)^$
+
+
+*/
+
+// non-compiled with javac: Compilable with Java25
+
+public static void main(String[] args) {
+    System.out.println("hello");
+}

